sdk/monitor/azure-monitor-query/LargeLogQuery.md

```diff
@@ -0,0 +1,123 @@
+
+## Getting Started
+
+An authenticated client is required to query data from Log Analytics. The following code shows how to create a `LogsQueryClient` using the
+`DefaultAzureCredential` authentication method.
+
+```java com.azure.monitor.query.LargeQuerySample-createLogsQueryClient
+LogsQueryClient client = new LogsQueryClientBuilder()
+    .credential(new DefaultAzureCredentialBuilder().build())
+    .buildClient();
+```
+
+### Setting the Log Analytics Workspace ID
+
+Set the `LOG_WORKSPACE_ID` variable to the ID of the Log Analytics workspace you want to query.
+
+```java com.azure.monitor.query.LargeQuerySample-setWorkspaceId
+static final String WORKSPACE_ID = "{workspace-id}";
+```
+
+### Defining the Helper Function
+
+In order to overcome the service limits, the strategy is to query data in smaller chunks based on some time column (i.e. TImeGenerated). The
+following helper function take a large query and splits it into a `LogsBatchQuery` based on a list of time ranges (called endpoints).
+
+```java com.azure.monitor.query.LargeQuerySample-createBatchQueryFromLargeQuery
+static LogsBatchQuery createBatchQueryFromLargeQuery(String originalQuery,
+                                                     List<OffsetDateTime> endpoints,
+                                                     String timeColumn) {
+    LogsBatchQuery batchQuery = new LogsBatchQuery();
+
+    for (int i = 0; i < endpoints.size(); i++) {
+        String queryString;
+        if (i == endpoints.size() - 1) {
+            queryString = String.format("%1$s | where %3$s >= datetime('%2$s')", originalQuery, endpoints.get(i), timeColumn);
+        }
+        else {
+            queryString = String.format("%1$s | where %4$s >= datetime('%2$s') and %4$s < datetime('%3$s')", originalQuery, endpoints.get(i), endpoints.get(i + 1), timeColumn);
+        }
+        batchQuery.addWorkspaceQuery(WORKSPACE_ID,queryString, null);
+    }
+
+    return batchQuery;
+}
+```
+
+### Defining the Query-BatchQuery conversion Functions
+
+This sample provids two functions to convert a `LogsQuery` into a `LogsBatchQuery`. One function splits the query into multiple smaller queries
+based upon the approximate* maximum amount of rows to be present in each query. The other function splits the query into multiple smaller queries based upon
+the approximate* maximum size of the data in bytes to be present in each query.
+
+_*If multiple rows have the exact same TimeGenerated, the queries in the batch query may exceed the maximum amount of rows/bytes_
+
+#### Splitting by Rows
+
+```java com.azure.monitor.query.LargeQuerySample-createBatchQueryFromLargeQueryByRows
+static LogsBatchQuery createBatchQueryFromLargeQueryByRowCount(String originalQuery,
+                                                               int maxRowPerBatch,
+                                                               String timeColumn) {
+
+    String findBatchEndpointsQuery = String.format(
```

Comment:
same here, would you mind adding an example with the original query and an example of formatted string?